src/sentry/integrations/github_enterprise/webhook.py

```diff
@@ -143,19 +145,30 @@ def handle(self, request):
             return HttpResponse(status=400)
 
         try:
-            # Attempt to validate the signature. Older versions of
-            # GitHub Enterprise do not send the signature so this is an optional step.
-            method, signature = request.META["HTTP_X_HUB_SIGNATURE"].split("=", 1)
-            if not self.is_valid_signature(method, body, secret, signature):
-                logger.warning(
-                    "github_enterprise.webhook.invalid-signature", extra=self.get_logging_data()
-                )
-                return HttpResponse(status=401)
-        except (KeyError, IndexError) as e:
-            logger.info(
-                "github_enterprise.webhook.missing-signature",
-                extra={"host": host, "error": str(e)},
+            try:
+                signature = request.META["HTTP_X_HUB_SIGNATURE_256"]
+            except KeyError:
+                signature = request.META["HTTP_X_HUB_SIGNATURE"]
+        except KeyError:
+            logger.error(
+                "github_enterprise.webhook.missing-signature", extra=self.get_logging_data()
+            )
+            return HttpResponse(status=400)
+
+        try:
+            method, signature = signature.split("=", 1)
+        except KeyError:
+            logger.error(
+                "github_enterprise.webhook.malformed-signature", extra=self.get_logging_data()
+            )
+            return HttpResponse(status=400)
+
+        if not self.is_valid_signature(method, body, self.get_secret(), signature):
```

Comment:
I think we just want `secret` here instead of `self.get_secret()` , this should get your tests passing as well